scenarios/kubernetes_e2e.py

```diff
@@ -257,14 +257,27 @@ def start(self, args):
         check_env(env, self.command, *args)
 
 
-def cluster_name(cluster, build):
+def cluster_name(cluster):
     """Return or select a cluster name."""
     if cluster:
         return cluster
-    # Create a suffix based on the build number. Append a random string to it for
-    # avoiding potential conflicts across different jobs' runs (see issue #7592).
-    suffix = build if len(build) < 10 else hashlib.md5(build).hexdigest()[:10]
-    return 'e2e-%s-%s' % (suffix, os.urandom(3).encode('hex'))
+    # Create a suffix based on the build number and job name.
+    # This ensures no conflict across runs of different jobs (see #7592).
+    # For PR jobs, we use PR number instead of build number to ensure the
+    # name is constant across different runs of the presubmit on the PR.
+    # This helps clean potentially leaked resources from earlier run that
+    # could've got evicted midway (see #7673).
+    job_type = os.getenv('JOB_TYPE')
+    if job_type == 'batch':
+        suffix = 'batch'
```

Comment:
Should we include PR numbers in the suffix to distinguish different batches or is that not necessary?

Reply:
That happens to be the case now, but you shouldn't make that assumption. We actually plan to update Tide to trigger multiple batches of different sizes like Steve suggests here: https://github.com/kubernetes/test-infra/issues/7561#issuecomment-378961503

Additionally, Tide has a testing pool for each branch (not the entire repo like the Submit Queue). This means that the job could be running as part of a batch for each different branch.

Reply:
further clarifying that: the submit-queue only batches master, in batches of <= 5 PRs. tide already batches every branch (simultaneously) and may have multiple batches of varying size and overlap. we should not even assume that refs are unique, build ID is more appropriate.

Reply:
Ben: Do you have some suggestions on how to deal with multiple parallel batches? One simple idea I have is to append a random string for 'batch' case to uniquify them.

Reply:
for any given job name every time we run the job we set `$BUILD_ID` to a unique ID for that run, this seems more appropriate than the refs to me (?) https://github.com/kubernetes/test-infra/tree/master/prow#job-environment-variables

Reply:
Not too sure, but from that link BUILD_ID seems to be same as BUILD_NUMBER.
Anyway, I've made the change to include that now.

Reply:
they are currently IIRC, for whatever reason BUILD_NUMBER is deprecated? /shrug
> Note: Use of $BUILD_NUMBER is deprecated. Please use $BUILD_ID instead.